test/utils/fs.ts

```diff
@@ -0,0 +1,13 @@
+import * as fs from 'fs/promises';
+
+export async function withTemporaryEdits<T = void>(
+  filePath: string,
+  doWork: () => Promise<T>,
+): Promise<T> {
+  const envOriginal = await fs.readFile(filePath, 'utf-8');
```

Comment:
This is not necessarily just for env files, right? So this variable could be called `originalFile`?